tsfresh/utilities/distribution.py

```diff
@@ -86,13 +89,18 @@ def partition(data, chunk_size):
         :rtype: generator
         """
 
-        iterable = iter(data)
-        while True:
-            next_chunk = list(itertools.islice(iterable, chunk_size))
-            if not next_chunk:
-                return
-
-            yield next_chunk
+        if isinstance(data, TsData):
+            return data.partition(chunk_size)
+        else:
+            def partition_iterable():
+                iterable = iter(data)
+                while True:
+                    next_chunk = list(itertools.islice(iterable, chunk_size))
```

Comment:
why do you name this variables `next_chunk`? you partition the whole data into one singular chunk here, right?

Reply:
This was already in place: https://github.com/blue-yonder/tsfresh/blob/29cdae3079fdd8f6dc50ad7eebfbf6e33f0a2b46/tsfresh/utilities/distribution.py#L91

But I think the name is ok. The function is yieldings slices of the iterator as list chunks.